tensorboard/examples/plugins/example_raw_scalars/tensorboard_plugin_example_raw_scalars/static/model.js

```diff
@@ -14,42 +14,48 @@
 // ==============================================================================
 
 // Generic example of a service that provides data.
-// In a production plugin, consider caching these results, to avoid making
-// unnecessary requests for data..
+
+/** @type {?Object<string, !Array<string>>} */
+let runToTagInfo = null;
+
+async function updateRunInfo() {
+  if (!runToTagInfo)
+    runToTagInfo = await fetchJSON('./tags') || {};
+}
 
 /**
- * @return {!Array<string>}
+ * @return {!Promise<?Array<string>>}
  */
 export async function getRuns() {
-  const runInfo = await fetchJSON('./runInfo');
-  return Object.keys(runInfo);
+  await updateRunInfo();
+  return Object.keys(runToTagInfo);
 }
 
 /**
  * @param {string} run
- * @return {!Array<string>}
+ * @return {!Promise<?Array<string>>}
```

Comment:
I believe `?Array` is saying the resolved type of Promise is either an Array or a null. Did you mean to say `Array|undefined`?

Reply:
Ah, really this function should always return an Array.  Updated.